src/com/photoncat/aiproj2/heuristic/Heuristic2.java

```diff
@@ -7,6 +7,20 @@
 public class Heuristic2 implements Heuristics {
     @Override
     public int heuristic(Board board){
+
+        //first check if there is win or draw
+        if(board.gameover()){
```

Comment:
This is wrong. Either side winning or draw all counts as a game over. You should put the code checking which side wins in the if branch instead of the else branch.